### PR TITLE
Creating relations

### PR DESCRIPTION
**Pull request by ankitjena**
## Motivation

Add the ability to reflect GraphQL schema relationships into database structure and resolvers. 

# Technical approach

TODO

## Tasks

- [ ] Database schema relationships
- [ ] Resolver generation for relationships.
- [ ] Documentation

## QA

figure out #7 . facing issues with one to many relations and how they will be created using knex.

### CODE REVIEW

**Comment by wtrocki** [comment]
And here we are. We hit the second issue related to not having the full scope of the data. Let's imagine the following scenario:

```
type Car {
  owners: [Person]
}


type Person {
  cars: [Car]
}
```

When processing single item we never know if the relationship is oneToMany or ManyToMany due to the fact that we process single type and we operate on it using knex. 

## Possible ways to solve this

Obviously, we can add relationship on the second type that will resolve `OneToMany` problem but we still do not know if we are `ManyToMany` relationship. 

To resolve this we can:

- Always assume `Many2Many` and create a separate relationship table 
- Use directive for steer behavior (Could be another step)
Something like this:

```
type Person {
  cars: [Car] @relation(Type="ManyToMany", table="PersonCars")
}

type Car {
  owners: [Person] @relation(Type="ManyToMany", table="PersonCars")
}
```

This information can be used to generate both resolvers and tables.

**Comment by ankitjena** [comment]
But that would mean creating a separate table for every relation won't it?

**Comment by wtrocki** [comment]
@ankitjena what you mean by 
> that

I would like to know your take on what I have written. 
What angle are we going to try here first?

**Comment by ankitjena** [comment]
The 2nd option seems good. Sorry I thought these were 2 steps. But knex provides support for foreign key creations, but I am unclear about creating many2many relations.

**Comment by wtrocki** [comment]
@ankitjena Perfect! We do not need to create many2many now. Let's create OneToMany using this directive. Then naturally we can create a separate table that will reference both and also create additional relationship type in the schema. 
Let's not work on all stuff at the moment and let's say we track progress using tasks:

- [ ] Create field using OneToMany relationship
- [ ] Create relationship table. 

Each of those can be separate commit I will review. 
If you stuck give me shout we can have some quick pair programming session.

**Comment by ankitjena** [comment]
Okay working on it.

File: example/Note.graphql [hunk 1]
```diff
@@ -4,7 +4,7 @@ type Note {
   title: String!
   description: String!
   ## Relationship
-  comment: [Comment]!
+  comment: [Comment]! @relation(type:"ManyToMany", table:"NoteComments")
```
Comment (wtrocki) [inline example/Note.graphql]: The directive is not defined. This is directive usage, but it needs to be defined.

**Pull request closed without merging**